org.mwc.cmap.TimeController/src/org/mwc/cmap/TimeController/recorders/CoordinateRecorder.java

```diff
@@ -144,14 +149,297 @@ protected void showMessageDialog(final String message)
+          new Debrief.ReaderWriter.Replay.ImportReplay());
+
+      // get our thread to import this
+      final MWC.Utilities.ReaderWriter.ImportManager.BaseImportCaller reader =
+          new MWC.Utilities.ReaderWriter.ImportManager.BaseImportCaller(
+              new File[]
+              {testFile}, _theLayers)
+          {
+            // handle completion of the full import process
+            @Override
+            public void allFilesFinished(final File[] fNames,
+                final Layers newData)
+            {
+              Application.logError2(Application.INFO,
+                  "All files loaded correctly", null);
+            }
+
+            // handle the completion of each file
+            @Override
+            public void fileFinished(final File fName, final Layers newData)
+            {
+              Application.logError2(Application.INFO, "File " + fName
+                  + " loaded correctly", null);
+            }
+          };
+
+      // and start it running
+      reader.start();
+
+      // wait for the results
+      while (reader.isAlive())
+      {
+        try
+        {
+          Thread.sleep(100);
+        }
+        catch (final java.lang.InterruptedException e)
+        {
+          Application.logError2(Application.INFO, "Error while sleeping", e);
+        }
+      }
+
+      /*
+       * final PlainProjection projection = new org.mwc.cmap.gt2plot.proj.GtProjection();
+       */
+      final PlainProjection projection =
+          new MWC.Algorithms.Projections.FlatProjection();
+      projection.setScreenArea(new java.awt.Dimension(1443, 901));
+      projection.setDataArea(new WorldArea(new WorldLocation(22.238965795584505,
+          -21.928244631862952, 0), new WorldLocation(22.238965795584505,
+              -21.43985414608609, 0)));
+
+      final org.mwc.cmap.core.DataTypes.Temporal.TimeControlProperties timePreferences =
+          new org.mwc.cmap.core.DataTypes.Temporal.TimeControlProperties();
+
+      CoordinateRecorder recorder = new CoordinateRecorder(_theLayers,
+          projection, timePreferences);
+
+      return recorder;
+    }
+
+    /**
+     * COLLINGWOOD starts after NELSON
+     */
+    public void testPrimaryStartsFirst()
+    {
+      CoordinateRecorder recorder = getRecorder();
+
+      MWC.GenericData.HiResDate currentTime = new MWC.GenericData.HiResDate(
+          818748540000L);
+
+      final int AMOUNT_OF_STEPS = 3;
+      doIteration(recorder, currentTime, AMOUNT_OF_STEPS);
+
+      final java.util.Map<String, Debrief.ReaderWriter.powerPoint.model.Track> track =
+          recorder._tracks;
+      checkTrackSize(track);
+      assertTrue(track.get(collingwood).getStepsToSkip() == 1);
+      assertTrue(track.get(collingwood).getSegments().size() == 2);
+      assertTrue("0064bd".equals(track.get(collingwood).getColorAsString()));
+      checkCollingwoodSegments(track);
+      assertTrue(track.get(nelson).getSegments().size() == 3);

```

Comment:
Aah, there's an issue from an earlier release here. A `segment` is a set of track points.  I'm pretty sure the above method should be `track.get(nelson).getPoints().size() == 3`.  So, the method (probably getter and setter) needs to be renamed to refer to points, not segments.